plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1753,205 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+
+        assertUpdate(session, "INSERT INTO " + tableName1 + "(x, part) values('a', 'part_a')", 1);
+        assertUpdate(session, "INSERT INTO " + tableName2 + "(x, part) values('a', 'part_a')", 1);
+
+        assertQueryFails(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.x = b.x) where a.x = 'a'".formatted(tableName1, tableName2), "Filter required on .*" + tableName1 + " for at least one partition column:.*");
+        assertQuery(session, "SELECT a.x, b.x from %s a JOIN %s b on (a.part = b.part) where a.part = 'part_a'".formatted(tableName1, tableName2), "VALUES ('a', 'a')");
+
+        assertUpdate(session, "DROP TABLE " + tableName1);
+        assertUpdate(session, "DROP TABLE " + tableName2);
+    }
+
+    @Test
+    public void testPartitionPredicateAllowed()
+    {
+        String tableName = "test_partition_predicate" + randomNameSuffix();
+        Session session = sessionWithPartitionFilterRequirement();
+        assertUpdate(session, "CREATE TABLE %s (x VARCHAR, part INTEGER) WITH (partitioned_by = ARRAY['part'])".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values('a', 1)", 1);
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values('b', 2)", 1);
+
+        assertQuery(session, "SELECT * FROM " + tableName + " WHERE CAST (part AS INTEGER) = 1", "VALUES ('a', 1)");
+        assertUpdate(session, "DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testPartitionPredicateInOuterQuery()
+    {
+        String tableName = "test_partition_predicate" + randomNameSuffix();
+        Session session = sessionWithPartitionFilterRequirement();
+        assertUpdate(session, "CREATE TABLE %s (x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(1, 11)", 1);
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(2, 22)", 1);
+
+        assertQuery(session, "SELECT * FROM (SELECT * FROM " + tableName + " WHERE CAST (x AS INTEGER) = 1) WHERE CAST(part AS INTEGER) = 11", "VALUES (1, 11)");
+        assertUpdate(session, "DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testPartitionPredicateInInnerQuery()
+    {
+        String tableName = "test_partition_predicate" + randomNameSuffix();
+        Session session = sessionWithPartitionFilterRequirement();
+        assertUpdate(session, "CREATE TABLE %s (x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])".formatted(tableName));
+
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(1, 11)", 1);
+        assertUpdate(session, "INSERT INTO " + tableName + "(x, part) values(2, 22)", 1);
+
+        assertQuery(session, "SELECT * FROM (SELECT * FROM " + tableName + " WHERE CAST (part AS INTEGER) = 11) WHERE CAST(x AS INTEGER) = 1", "VALUES (1, 11)");
```

Comment:
`CAST` is redundant?

Reply:
Yes, from the formal point is redundant. However if I took other connector tests as reference , it was also considered as  a testcase.